### PR TITLE
ECR Builds

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Adds ECR Builds

## why
* Occasionally get dockerhub rate limited, public ECR fixes that

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -10,12 +10,31 @@ on:
 jobs:
   build-and-push:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      IAM_ROLE_SESSION_NAME: geodesic-ci
+      AWS_REGION: us-east-1
+      ECR_REGISTRY: public.ecr.aws/h1u4b2i4/
```
Comment (osterman) [inline .github/workflows/docker.yml:19]: Is this customizable?

Comment (Benbentwo) [reply .github/workflows/docker.yml:19]: Yes but needs to get approved by amazon, put in request a while ago, might need some verification

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -10,12 +10,32 @@ on:
 jobs:
   build-and-push:
     runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      IAM_ROLE_SESSION_NAME: geodesic-ci
+      AWS_REGION: us-east-1
+      ECR_REGISTRY: public.ecr.aws/ # This will become `public.ecr.aws/cloudposse/$repositoryName`
```
Comment (osterman) [inline .github/workflows/docker.yml:19]:
```suggestion
      ECR_REGISTRY: public.ecr.aws/ # Images will be published to `public.ecr.aws/cloudposse/$repositoryName`
```

**Review by osterman — APPROVED** [review]

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -10,12 +10,32 @@ on:
...
+      with:
+        role-to-assume: ${{ secrets.ECR_AWS_ROLE }}
+        aws-region: ${{ env.AWS_REGION }}
+        role-session-name: ${{ env.IAM_ROLE_SESSION_NAME }}
+
+    - name: Login to Public Amazon ECR
+      run: |
+        docker logout public.ecr.aws
+        aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+
```
Comment (nitrocode) [inline .github/workflows/docker.yml:38]: You could also add the metadata action

```suggestion

      - name: Docker meta
        id: meta
        uses: docker/metadata-action@v4
        with:
          images: cloudposse/geodesic
```

Comment (Benbentwo) [reply .github/workflows/docker.yml:38]: I propose we make this update in a future PR similar to @osterman's comment below

Comment (nitrocode) [reply .github/workflows/docker.yml:38]: sure but the tagging is going to be much more complicated than adding the labels. I would recommend 2 separate PRs. 

1. PR to add labels and metadata action
2. PR to modify the complicated tags logic below to use metadata output

**Pull request merged**